### PR TITLE
lib/promscrape: respect `0` value for `series_limit` param

### PR DESCRIPTION
**Pull request by hagen1778**
Respect `0` value for `series_limit` param in `scrape_config` even if global limit was set via `-promscrape.seriesLimitPerTarget`. Previously, `0` value will be ignored in favor of `-promscrape.seriesLimitPerTarget`.

This behavior aligns with possibility to override `series_limit` value via relabeling with `__series_limit__` label.

### CODE REVIEW

**Review by Haleygo — APPROVED** [review]
LGTM

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -76,6 +76,7 @@ The sandbox cluster installation is running under the constant load generated by
...
 * BUGFIX: [vmagent](https://docs.victoriametrics.com/vmagent.html): properly discover targets for `role: endpoints` and `role: endpointslice` in [kubernetes_sd_configs](https://docs.victoriametrics.com/sd_configs.html#kubernetes_sd_configs). Previously some `endpoints` and `endpointslice` targets could be left undiscovered or some targets could have missing `__meta_*` labels when performing service discovery in busy Kubernetes clusters with large number of pods. See [this pull request](https://github.com/VictoriaMetrics/VictoriaMetrics/pull/5557).
+* BUGFIX: [vmagent](https://docs.victoriametrics.com/vmagent.html): respect `0` value for `series_limit` param in [scrape_config](https://docs.victoriametrics.com/sd_configs.html#scrape_configs) even if global limit was set via `-promscrape.seriesLimitPerTarget`. Previously, `0` value will be ignored in favor of `-promscrape.seriesLimitPerTarget`.
```
Comment (valyala) [inline docs/CHANGELOG.md:79]:
```suggestion
* BUGFIX: [vmagent](https://docs.victoriametrics.com/vmagent.html): respect explicitly set `series_limit: 0` in [scrape_config](https://docs.victoriametrics.com/sd_configs.html#scrape_configs). This allows removing [`series_limit` restriction](https://docs.victoriametrics.com/vmagent.html#cardinality-limiter) on a per-`scrape_config` basis when global limit is set via `-promscrape.seriesLimitPerTarget`. Previously, `0` value was ignored in favor of `-promscrape.seriesLimitPerTarget`.
```

**Review by valyala — APPROVED** [review]

**Comment by valyala** [comment]
@hagen1778 , thanks for the fix!

**Comment by valyala** [comment]
@hagen1778 , see also the follow-up commit fac7c30f4ee7428988a3b21cec68ab2fc2580a6d , which updates [the docs regarding `series_limit`](https://docs.victoriametrics.com/vmagent.html#cardinality-limiter)

**Pull request merged**